test/literal_test.cc

```diff
@@ -383,4 +383,211 @@ TEST(LiteralTest, DoubleZeroComparison) {
+            fixed(8)},
+        LiteralRoundTripParam{
+            "FixedLength16",
+            {0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C,
+             0x0D, 0x0E, 0x0F},
+            Literal::Fixed({0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09,
+                            0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F},
+                           16),
+            fixed(16)},
+        LiteralRoundTripParam{
+            "FixedSingleByte", {0xFF}, Literal::Fixed({0xFF}, 1), fixed(1)},
+
+        // Temporal types
+        LiteralRoundTripParam{"DateEpoch", {0, 0, 0, 0}, Literal::Date(0), date()},
+        LiteralRoundTripParam{"DateNextDay", {1, 0, 0, 0}, Literal::Date(1), date()},
+        LiteralRoundTripParam{"DateY2K", {205, 42, 0, 0}, Literal::Date(10957), date()},
+        LiteralRoundTripParam{
+            "DateNegative", {255, 255, 255, 255}, Literal::Date(-1), date()},
+
+        LiteralRoundTripParam{
+            "TimeMidnight", {0, 0, 0, 0, 0, 0, 0, 0}, Literal::Time(0), time()},
+        LiteralRoundTripParam{"TimeNoon",
+                              {128, 9, 230, 124, 10, 0, 0, 0},
+                              Literal::Time(45045123456),
+                              time()},
+        LiteralRoundTripParam{
+            "TimeOneSecond", {64, 66, 15, 0, 0, 0, 0, 0}, Literal::Time(1000000), time()},
+
+        LiteralRoundTripParam{"TimestampEpoch",
+                              {0, 0, 0, 0, 0, 0, 0, 0},
+                              Literal::Timestamp(0),
+                              timestamp()},
+        LiteralRoundTripParam{"TimestampOneSecond",
+                              {64, 66, 15, 0, 0, 0, 0, 0},
+                              Literal::Timestamp(1000000),
+                              timestamp()},
+        LiteralRoundTripParam{"TimestampNoon2024",
+                              {128, 9, 230, 124, 10, 0, 0, 0},
+                              Literal::Timestamp(45045123456),
+                              timestamp()},
+
+        LiteralRoundTripParam{"TimestampTzEpoch",
+                              {0, 0, 0, 0, 0, 0, 0, 0},
+                              Literal::TimestampTz(0),
+                              timestamp_tz()},
+        LiteralRoundTripParam{"TimestampTzOneHour",
+                              {0, 164, 147, 214, 0, 0, 0, 0},
+                              Literal::TimestampTz(3600000000),
+                              timestamp_tz()}),
+
+    [](const testing::TestParamInfo<LiteralSerializationParam::ParamType>& info) {
+      return info.param.test_name;
+    });
+
+TEST(LiteralSerializationTest, EmptyString) {
+  auto empty_string = Literal::String("");
+  auto empty_bytes = empty_string.Serialize();
+  ASSERT_TRUE(empty_bytes.has_value());
+  EXPECT_TRUE(empty_bytes->empty());
+
+  auto deserialize_result = Literal::Deserialize(*empty_bytes, string());
+  EXPECT_THAT(deserialize_result, IsError(ErrorKind::kInvalidArgument));
```

Comment:
Could you double check the spec and java impl on how to deal with empty values? I think this is a valid case which shouldn't return an error.